test/jdk/java/awt/TextArea/TextAreaCursorTest/HoveringAndDraggingTest.java

```diff
@@ -21,71 +21,76 @@
 import java.awt.TextArea;
-import java.awt.Dialog;
 
-public class HoveringAndDraggingTest extends java.applet.Applet {
-    public void start() {
-        String[] instructions = new String[] {
-            "1. Notice components in test window: main-panel, box-for-text,"
-                +" 2 scroll-sliders, and 4 scroll-buttons.",
-            "2. Hover mouse over box-for-text."
-                +" Make sure, that mouse cursor is TextCursor (a.k.a. \"beam\").",
-            "3. Hover mouse over each of components (see item 1), except for box-for-text."
-                +" Make sure, that cursor is DefaultCursor (arrow).",
-            "4. Drag mouse (using any mouse button) from box-for-text to every"
-                +" component in item 1, and also outside application window."
-                +" Make sure, that cursor remains TextCursor while mouse button is pressed.",
-            "5. Repeat item 4 for each other component in item 1, except for box-for-text,"
-                +" _but_ now make sure that cursor is DefaultCursor.",
-            "6. If cursor behaves as described in items 2-3-4-5, then test passed; otherwise it failed."
-        };
-        Sysout.createDialogWithInstructions( instructions );
+/*
+ * @test
+ * @bug 6497109
+ * @summary Mouse cursor icons for TextArea should be correct in case of
+ *  hovering or dragging mouse over different subcomponents.
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual HoveringAndDraggingTest
+ */
+
+public class HoveringAndDraggingTest {
+    public static void main(String[] args) throws Exception {
+        String INSTRUCTIONS = """
+                    1. Notice components in test window: main-panel, box-for-text,
+                       2 scroll-sliders, and 4 scroll-buttons(Not applicable for macosx).
+                    2. Hover mouse over box-for-text.
+                       Make sure, that mouse cursor is TextCursor (a.k.a. "beam").
+                    3. Hover mouse over each of components (see item 1), except for box-for-text.
+                       Make sure, that cursor is DefaultCursor (arrow).
+                    4. Drag mouse (using any mouse button) from box-for-text to every
+                       component in item 1, and also outside application window.
+                       Make sure, that cursor remains TextCursor while mouse button is pressed.
+                    5. Repeat item 4 for each other component in item 1, except for box-for-text
+                       _but_ now make sure that cursor is DefaultCursor.
+                    6. If cursor behaves as described in items 2-3-4-5, then test passed;
+                       otherwise it failed.
+                """;
+
+        PassFailJFrame.builder()
+                .title("Test Instructions")
+                .instructions(INSTRUCTIONS)
+                .rows((int) INSTRUCTIONS.lines().count() + 2)
+                .columns(40)
+                .testUI(initialize())
```

Comment:
```suggestion
                .testUI(HoveringAndDraggingTest::initialize)
```
You should pass a method reference for the `PassFailJFrame` framework to create UI components on EDT, unless you need to create UI components on the main thread. See [this comment explaining the difference](https://github.com/openjdk/jdk/pull/21029#discussion_r1764872438).

Reply:
Yes, understood.